src/gluonts/dataset/loader.py

```diff
@@ -168,6 +193,9 @@ def __init__(
         num_prefetch: Optional[int] = None,
         decode_fn: Callable = lambda x: x,
     ) -> None:
+        assert num_workers is None or num_workers > 0
```

Comment:
There is `core.ty.checked`, which we could also use for these assertions.

Reply:
should I use pydantic types to enforce positiveness?